examples/agentInteractionsDemo/app.js

```diff
@@ -175,13 +175,88 @@ new Vue({
         const qualityApi = new platformClient.QualityApi();
         const conversationsApi = new platformClient.ConversationsApi();
+        const externalContactsApi = new platformClient.ExternalContactsApi();
         Vue.prototype.$usersApi = usersApi;
         Vue.prototype.$qualityApi = qualityApi;
         Vue.prototype.$conversationsApi = conversationsApi;
+        Vue.prototype.$externalContactsApi = externalContactsApi;
 
         let authenticated = false;
         let agentUserId = "";
 
+        function getCustomerParticipant(conv) {
+            let newConv = conv;
+            newConv.customer = newConv.participants.find((part) => {
+                return part.purpose === "customer" || part.purpose === "external";
+            }) || { name: "Unknown" };
+            return newConv.customer
+        }
+
+        async function getCustomerName(customer){
+            if (customer.externalContactId){
+                return externalContactsApi.getExternalcontactsContact(customer.externalContactId)
+                    .then((externalContact) => {
+                        const name = `${externalContact.firstName} ${externalContact.lastName}`;
+                        return Promise.resolve(name)
+                    })
+            } else{
+                if(!customer.name){
+                    return Promise.resolve("Unknown")
+                } else{
+                    return Promise.resolve(customer.name)
+                }
+            } 
+        }
+
+        async function getConversationData(convId){
+            return conversationsApi.getConversation(convId).then(
+                    async (conv) => {
+                        conv.customer = getCustomerParticipant(conv);
+                        const name = await getCustomerName(conv.customer);
+                        conv.customer.name = name;
+                        return conv;
+                    }
+                ).catch((err) => {
+                    console.log(`Error: ${err}`);
+                })
+        }
+
+        async function getConversationsAndEvaluations(agentUserId) {
+            const startTime = moment('1997-01-01').toISOString();
+            const endTime = moment().toISOString();
+            return getEvaluations(qualityApi, startTime, endTime, agentUserId)
+                .then(async (data) => {
+                    const evaluations = data.entities;
+                    const evalConversations = {}
+
+                    return Promise.all(
+                        evaluations.map(async eval => {
+                            const evalConvId = eval.conversation.id
+                            if (!(evalConvId in evalConversations)){
+                                evalConversations[evalConvId] = {evals: []}
+                            }
+                            evalConversations[evalConvId].evals.push(eval)
+                            const evalConv = await getConversationData(evalConvId);
```

Comment:
looks like getConversationData catches, logs, and returns undefined.

you have a few options imo,
* check for undefined and don't add the eval or convo to the list
* let it throw, change `Promise.all` to `Promise.allSettled` and then filter out evals with invalid convos
* allow the convo to be undefined and handle it in the template.

I'm basing that